src/digest.rs

```diff
@@ -22,25 +22,29 @@ pub trait HasDigest<F: PrimeField> {
 
 /// Trait for components with potentially discrete digests to be included in their container's digest.
 pub trait Digestible {
-  /// Write the byte representation of Self in a byte buffer
-  fn write_bytes<W: Sized + io::Write>(&self, byte_sink: &mut W) -> Result<(), io::Error>;
+  /// Write the digest representation of Self in a byte buffer
+  /// this is what you want to write when Self is being digested into another digest
+  fn write_digest<W: Sized + io::Write>(&self, byte_sink: &mut W) -> Result<(), io::Error>;
```

Comment:
I think this may need a better name. I think the separated functions make sense, but for example `SimpleDigestible` implements this function in a way that does not involve writing it's *digest*.

Maybe this can be called `write_summary_bytes` or something? That is not a well-thought-out suggestion, but the point is that this function might either write the digest or the input over which the digest was created — depending on context. At least that is what I've understood.

Reply:
That does make sense; maybe `write_digestable_bytes` to indicate that these are the bytes being digested (into another container)?

Reply:
Yes, I think that's the ticket.